articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,744 @@
+---
+layout: default
+title: ROS 2 Launch XML Format
+permalink: articles/roslaunch_xml.html
+abstract:
+  The XML format for declarative launch descriptions in the ROS 2 launch system.
+author: '[Michel Hidalgo](https://github.com/hidmic)'
+published: false
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Authors: {{ page.author }}
+
+
+# ROS 2 Launch XML Format v0.1.0
+
+## Rationale
+
+As an alternative to a programmatic approach to the ROS 2 launch system's API, a declarative description features a WYSIWYG approach, easier to read, audit and maintain.
+This is the preferred approach for ROS 1 `roslaunch` launch files, thus some degree of familiarity is expected (and relied upon).
```

Comment:
There's a section in the (still open) pr for launch's design document for this topic. It might require less context if this were part of that document.

Additionally, I'd like to see this in two parts:

- a way for any markup language to map to the Python API in launch
  - and specifically how that could be done for XML, e.g. something explicit but not necessarily convenient like `<launch_action pkg="launch_ros" class="Node"><launch_action_arg type="int">42</launch_action_arg>...</launch_action>`
- syntactic sugar that builds on the aforementioned explicit matching, e.g. `<node ...>` rather than `<launch_action pkg="launch_ros" class="Node"...` etc

I have enough thoughts about this and how it might be implemented to probably warrant a meeting to discuss it. At the very least we should discuss to see if it's worth taking this approach to the implementation versus a more direct implementation which doesn't worry about an automatic mapping of any type.

Reply:
The idea was to have a very brief intro, followed by a link to the actual architecture design. That made sense because, in my mind, one thing is the launch's architecture design document and another thing is a format specification for one of its front-ends. I wouldn't collapse them all into a single page. 

In that context, I think that implementation hints for XML parsing should go here, while implementation hints for a generalized approach to mapping declarative descriptions to programmatic ones should go in launch's design document. 

We can meet to talk about the latter, though I do see value on having this format spec regardless of how we end up implementing the mapping (see replies above).